### ISSUE TITLE
Misleading error message in Flatten

### ISSUE THREAD

**Issue by tsnowlan** [brentp/vcfanno#118]
Related to: #82 

Config and files used are arbitrary, just make a typo in the path for `file` in config toml and `-base-path`.

When attempting to open a Source file, the basepath is prepended twice. The first time in [Sources](https://github.com/brentp/vcfanno/blob/8ffa89c1/shared/shared.go#L99-L102), the next time in [Flatten](https://github.com/brentp/vcfanno/blob/8ffa89c1/shared/shared.go#L56-L59). This causes the error string to be `[Flatten] unable to open file: basepath/basepath/filename in basepath`. which implies that the basepath string was doubled somewhere rather than `basepath/filename` also not existing.

PR incoming with proposed fix.

---

If you have encountered an error, please include:

- [x] minimal conf and lua files that you are using.
- [x] urls or actual files for annotations in conf file.
- [x] minimal query file.
- [x] the command you used to invoke vcfanno
- [x] the **full** error message

**Comment by tsnowlan** [comment]
For the sake of completeness, from `example/conf.toml`: 
using [bad_conf.txt](https://github.com/brentp/vcfanno/files/4402044/bad_conf.txt) (github doesn't like .toml extension)

```
~/vcfanno$ vcfanno -base-path example -lua custom.lua example/bad_conf.txt example/query.vcf.gz

=============================================
vcfanno version 0.3.2 [built with go1.13.1]

see: https://github.com/brentp/vcfanno
=============================================

ERROR: can't find query file: example/example/query.vcf.gz

```

**Comment by brentp** [comment]
closed in #119
